frontend/packages/console-app/src/components/quick-starts/utils/quick-start-context.ts

```diff
@@ -187,6 +188,23 @@ export const useValuesForQuickStartContext = (): QuickStartContextValues => {
     [setAllQuickStartStates],
   );
 
+  const getQuickStartActiveTask = useCallback(
+    (totalTasks) => {
+      let activeTaskNumber = 0;
+      setAllQuickStartStates((qs) => {
+        const quickStart = qs[activeQuickStartID];
+        while (
+          activeTaskNumber !== totalTasks &&
+          quickStart[`taskStatus${activeTaskNumber}`] !== QuickStartTaskStatus.INIT
+        ) {
+          activeTaskNumber++;
+        }
+      });
```

Comment:
Why are you calling the setState here and you are not returning a new state as well?
looks like you just want to loop over All the quickstarts and find the active taskNumber for an active quick start.
You can just use the activeQuickStartState for that.